### DISCUSSION TITLE
cron jobs timing

### DISCUSSION THREAD

**Discussion by dimitarnikolovv** [pocketbase/pocketbase#3771] [Q&A] [answered]
I am facing a bit of an issue configuring the cron jobs on our server. The first issue I am facing is that even though our server's system and hardware clocks are configured to a European time zone the cron jobs are executing in UTC, which is not that big of a deal. The main problem is that when executing time-dependent queries, I am not exactly sure what 
`new Date().toISOString()` outputs.

An example that runs at 03:30:
```js
cronAdd('mark_expired_posts', '30 1 * * *', () => {
  console.log('MARKING EXPIRED POSTS!');
  const posts = arrayOf(
    new DynamicModel({
      id: '',
    })
  );

  $app
    .dao()
    .db()
    .select('id')
    .from('posts')
    .where(
      $dbx.exp('departDateTime < {:date}', {
        date: new Date().toISOString(),
      })
    )
    .all(posts);

  for (let i = 0; i < posts.length; i++) {
    const post = $app.dao().findRecordById('posts', posts[i].id);
    post.set('expired', true);
    $app.dao().saveRecord(post);
  }
});
```

However, if a post has a `departDateTime` such as `2023-11-17 05:30:00` **(UTC)** it gets marked by the function, bare In mind that the mark must be datetime-dependent, not only date-dependent, and that's where the issue comes from.

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase dates are not ISO8601 but RFC3339 compatible (_this is primary for backward compatibility reasons with earlier versions and may change in the future once custom filter dates format is supported_).

Or in other words, you can try to convert your date to RFC3339 compatible one with something like `new Date().toISOString().replace("T", " ")` (_aka. replace the `T` delimiter with a single space character_).

---

Alternatively and depending on your use case, you may not actually need a cron job for this. I don't know what is the meaning of `departDateTime`, but you can avoid manually marking the posts as "expired" and instead add a `expiredDate` field that can be preset in a before create hook with the expected date when the post will no longer be considered valid.
Then you can filter "non-expired" posts by doing something like: `expiredDate > @now`.
To prevent users from manually modifying the field value you can use update/create API rule like: `@request.data.expiredDate:isset = false`.
